jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/JaxRsResourceBase.java

```diff
@@ -533,4 +535,48 @@ protected void verifyNumberOfElements(int actual, int expected, String errorMess
         Preconditions.checkArgument(actual == expected, errorMessage);
     }
 
+    protected Response createPaymentResponse(final UriInfo uriInfo, final Payment payment, final TransactionType transactionType, @Nullable final String transactionExternalKey) {
+        final PaymentTransaction createdTransaction = findCreatedTransaction(payment, transactionType, transactionExternalKey);
+        Preconditions.checkNotNull(createdTransaction, "No transaction of type '%s' found", transactionType);
```

Comment:
That's actually a tricky one: Does it fail because client passed the wrong `transactionType` (=> `404`) or because our state is inconsistent (=> `5xx`). It would make sense to assume the former, but this is so not obvious.

Also it would be nice to use some kind of Preconditions in which case we might need to tweak the mapping into http status?